### PR TITLE
Update docs about CC3D firmware flashing and updating

### LINKED ISSUE #1617: Drop "OpenPilot" compatible mode from CC3D board documentation

### PR DESCRIPTION
**Pull request by andrey-skl**
FIxes #1617 
Please message me if any serious english grammar issues. I'm not native so I would like to improve the docs but can't do this without english mistakes.

### CODE REVIEW

**Comment by huston007** [comment]
@stronnag I believe one can, but I didn't try it. DId you try? If it 
Could you please tell if there is known default command line command to run it to flash CC3D?

**Comment by stronnag** [comment]
On posix operating systems (Linux, OSX), the following works fine for 'rescue' (i.e. shorting boot pins):
````
#!/bin/bash

SPEED=${1:-115200}
stty -F /dev/ttyUSB0 raw speed $SPEED -crtscts cs8 -parenb -cstopb -ixon
stm32flash -w cleanflight/obj/cleanflight_CC3D.hex -v -g 0x0 -b $SPEED /dev/ttyUSB0
````
and normally (same as) `make TARGET=CC3D flash`:

````
#!/bin/bash

SPEED=${1:-115200}
stty -F /dev/ttyUSB0 raw speed $SPEED -crtscts cs8 -parenb -cstopb -ixon
echo -n 'R' >/dev/ttyUSB0
stm32flash -w cleanflight/obj/cleanflight_CC3D.hex -v -g 0x0 -b $SPEED /dev/ttyUSB0
````

The transmission of the 'R' character is what switches CF into boot loader mode (which is what the configurator does). See also https://sourceforge.net/projects/stm32flash/

(so if you add something along the above, you have a solution across pretty much all OS)

**Comment by hydra** [comment]
The flashing instructions are pretty much the same for each target apart from the port/cable used. I feel the CC3D flashing instructions should just reference and link to generic USB->USART flashing instructions.

**Comment by huston007** [comment]
@hydra I'm sorry but I didn't find the place where flashing instructions already located. [This place](https://github.com/cleanflight/cleanflight/blob/master/docs/USB%20Flashing.md#usb-flashing) looks similar but there are no such instructions about stm32flas usage.

**Comment by hydra** [comment]
@huston007 Yes, you could rename that file to 'Flashing.md' and then add a section for generic flashing instructions.

File: docs/Board - CC3D.md [hunk 1]
```diff
@@ -103,28 +103,58 @@ You cannot use USART3 and I2C at the same time.
...
 # Flashing
 
-There are two primary ways to get Cleanflight onto a CC3D board.
+Since 1.11.0 release there is only one way to get Cleanflight onto a CC3D board: flash it to board using an FTDI USB-TTL converter and special software:
+* Windows only: (STM Flash Loader Demonstrator)[http://www.st.com/web/en/catalog/tools/PF257525]
+* Windows/Linux/OS X: (stm32flash utility)[http://sourceforge.net/projects/stm32flash/]. Here is a (manual online)[http://manpages.ubuntu.com/manpages/wily/man1/stm32flash.1.html]
 
-* Single binary image mode - best mode if you don't want to use OpenPilot.
-* OpenPilot Bootloader compatible image mode - best mode if you want to switch between OpenPilot and Cleanflight.
+Please not that some user have issues with flashing CC3D via non-FTDI adapter. In that case "STM Flash Loader Demonstrator" just cannot connect to the target. Also, Arduino's embedded FTDIs may not work too.
```
Comment (reedbn) [inline docs/Board - CC3D.md]: Please not**e** that some user have issues with flashing CC3D via non-FTDI adapter. In that case "STM Flash Loader Demonstrator" just cannot connect to the target. ~~Also,~~ Arduino's embedded FTDIs may **also** not work ~~too~~.

File: docs/Board - CC3D.md [hunk 2]
```diff
@@ -103,28 +103,58 @@ You cannot use USART3 and I2C at the same time.
...
 
-## Single binary image mode.
+The image can be flashed by an USB to UART adapter connected to the main port when the CC3D is put into the STM32 bootloader mode by powering on the CC3D with the SBL/3.3v pads bridged. You can than upgrade cleanflight with USB-Serial adapter via Cleanflight configurator without shorting boot pins.
```
Comment (reedbn) [inline docs/Board - CC3D.md]: You can th~~a~~**e**n upgrade

Comment (huston007) [inline docs/Board - CC3D.md]: Thanks, I pushed a fix.

**Comment by raydesheles** [comment]
Please help I have a r t s 250 yks I flew it one time then I was going to rearrange everything because it was not placed properly I put everything back and now it won't start up I believe because I have missed the buttons up on my FlySky I6 controller Flash but how do I download cc3d cleanflight for the OTG cable Android phone to controller with because I do not have a computer

**Comment by raydesheles** [comment]
Oh my email/ raydesheles985@gmail.com

**Pull request closed without merging**